### PR TITLE
baseholder implementation for timeouts

### PR DESCRIPTION
**Pull request by BobDotCom**
## Summary

Implements timeouts

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [x] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

File: discord/member.py [hunk 1]
```diff
@@ -266,6 +270,7 @@ class Member(discord.abc.Messageable, _UserTag):
         '_user',
         '_state',
         '_avatar',
+        'communication_disabled_until'
```
Comment (Dorukyum) [inline discord/member.py:273]:
```suggestion
        'communication_disabled_until',
```

File: discord/member.py [hunk 2]
```diff
@@ -645,26 +654,29 @@ async def edit(
...
+        +------------------------------+--------------------------------------+
+        | mute                         | :attr:`Permissions.mute_members`     |
+        +------------------------------+--------------------------------------+
+        | deafen                       | :attr:`Permissions.deafen_members`   |
+        +------------------------------+--------------------------------------+
+        | roles                        | :attr:`Permissions.manage_roles`     |
+        +------------------------------+--------------------------------------+
+        | voice_channel                | :attr:`Permissions.move_members`     |
+        +------------------------------+--------------------------------------+
+        | communication_disabled_until | :attr:`Permissions.kick_members`     |
```
Comment (Lulalaby) [inline discord/member.py:678]:
```suggestion
        | communication_disabled_until | :attr:`Permissions.manage_members`     |
```

File: discord/member.py [hunk 3]
```diff
@@ -748,10 +764,64 @@ async def edit(
...
         if payload:
             data = await http.edit_member(guild_id, self.id, reason=reason, **payload)
             return Member(data=data, guild=self.guild, state=self._state)
 
+    async def timeout(self, until: Optional[datetime.datetime], reason: str = None) -> None:
+        """|coro|
+
+        Timeouts a member from the guild for the set duration.
+
+        You must have the :attr:`~Permissions.kick_members` permission to
```
Comment (Lulalaby) [inline discord/member.py:782]:
```suggestion
        You must have the :attr:`~Permissions.manage_members` permission to
```

File: discord/member.py [hunk 4]
```diff
@@ -748,10 +764,64 @@ async def edit(
...
+            An error occurred doing the request.
+        """
+        await self.edit(communication_disabled_until=until, reason=reason)
+
+    async def remove_timeout(self, reason: str = None) -> None:
+        """|coro|
+
+        Removes the timeout from a member.
+
+        You must have the :attr:`~Permissions.kick_members` permission to
```
Comment (Lulalaby) [inline discord/member.py:806]:
```suggestion
        You must have the :attr:`~Permissions.manage_members` permission to
```

**Review by Lulalaby — CHANGES_REQUESTED** [review]

**Comment by BobDotCom** [comment]
Going to see if this needs any changes for the new timeout limits on the api

**Comment by BobDotCom** [comment]
@Lulalaby your changes may have broken the table in the last commit.

**Comment by Lulalaby** [comment]
Which table

**Comment by Lulalaby** [comment]
Oh

File: discord/member.py [hunk 5]
```diff
@@ -645,26 +654,29 @@ async def edit(
...
+        +------------------------------+--------------------------------------+
+        | mute                         | :attr:`Permissions.mute_members`     |
+        +------------------------------+--------------------------------------+
+        | deafen                       | :attr:`Permissions.deafen_members`   |
+        +------------------------------+--------------------------------------+
+        | roles                        | :attr:`Permissions.manage_roles`     |
+        +------------------------------+--------------------------------------+
+        | voice_channel                | :attr:`Permissions.move_members`     |
+        +------------------------------+--------------------------------------+
+        | communication_disabled_until | :attr:`Permissions.manage_members`     |
```
Comment (Lulalaby) [inline discord/member.py:678]:
```suggestion
        | communication_disabled_until | :attr:`Permissions.manage_members`   |
```

**Review by Lulalaby — CHANGES_REQUESTED** [review]

**Comment by Lulalaby** [comment]
Fixed 😘

**Comment by Lulalaby** [comment]
Timeouts are now locked to staff

**Comment by BobDotCom** [comment]
> Timeouts are now locked to staff

Timeouts have now been unlocked and enabled on all Partner and DMD servers.

**Review by Dorukyum — APPROVED** [review]

**Review by RPSMain — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**